#!/usr/bin/env python
PACKAGE = "hrsi_representation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("quantisation_factor", double_t, 0, "The quintisation for the creation of 0-satest in m",       .1, 0,    10)
gen.add("distance_threshold",  double_t, 0, "The distance threshold for qtcbc transitions in m",     122.0, 0,    10)
gen.add("smoothing_rate",      double_t, 0, "The smoothing rate in s",                                  .3, 0,    10)
gen.add("validate",            bool_t,   0, "Only create valid transtions",                           True)
gen.add("no_collapse",         bool_t,   0, "Do not collapse similar adjacent states. If qtc_type is set to qtcbc this is always True.",  False)

qtc_type_enum = gen.enum([ gen.const("qtcb",     int_t, 0, "1D QTC_B"),
                           gen.const("qtcc",     int_t, 1, "2D QTC_C"),
                           gen.const("qtcbc",    int_t, 2, "1D/2D QTC_BC")],
                         "An enum to set the target qtc type")

gen.add("qtc_type", int_t, 0, "The target QTC type", 1, 0, 2, edit_method=qtc_type_enum)

exit(gen.generate(PACKAGE, "hrsi_representation", "OnlineQTCCreator"))
